### PR TITLE
feat(brewfather): brewfather segment work squashed

### PR DESCRIPTION
**Pull request by memcpy-rand-rand-rand**
segment code and docs
modification to doGet to support request modifier funcs

### Prerequisites

- [X] I have read and understood the `CONTRIBUTING` guide
- [X] The commit message follows the [conventional commits][cc] guidelines
- [X] Tests for the changes have been added (for bug fixes/features)
- [X] Docs have been added / updated (for bug fixes/features)

feat(brewfather): show brew batch information from Brewfather's REST API

new segment code and docs
additional functionality on env.doGet to add request modifiers

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segment_brewfather.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+	return bf.props.getString(BFFlatIcon, "→")
+}
+
+func (bf *brewfather) getBatchStatusIcon(batchStatus string) string {
+	switch batchStatus {
+	case BFStatusPlanning:
+		return bf.props.getString(BFPlanningStatusIcon, "")
+	case BFStatusBrewing:
+		return bf.props.getString(BFBrewingStatusIcon, "")
```
Comment (JanDeDobbeleer) [inline src/segment_brewfather.go]: the defaults need the `\uxxxx` notation

File: src/segment_brewfather.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+		bf.URL = fmt.Sprintf("https://web.brewfather.app/tabs/batches/batch/%s", batchID)
+	}
+
+	bf.DefaultString = bf.formatDefaultString()
+
+	return true
+}
+
+// build a default string for the segment using Sprintf for speed and to provide a simple starting point without complex template logic
+func (bf *brewfather) formatDefaultString() string {
```
Comment (JanDeDobbeleer) [inline src/segment_brewfather.go]: You could also add default template? I want to move away from the string based implementations. I see you have both but it doesn't seem like `DefaultString` is used anywhere, right? If it's up to me, I would not have this function and only rely on the default template.

File: src/segment_brewfather.go (hunk 1 above)
Comment (memcpy-rand-rand-rand) [reply src/segment_brewfather.go]: That makes more sense than the evil glyph boxes of doom!

File: src/environment.go [hunk 3]
```diff
@@ -56,6 +56,8 @@ type cache interface {
 	set(key, value string, ttl int)
 }
 
+type doGetRequestModifier func(request *http.Request)
```
Comment (JanDeDobbeleer) [inline src/environment.go]: I like the idea, would rename to `HTTPRequestModifier`

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
I love this addition. It almost too crazy to believe that there's a brewing segment. Just a few remarks, let's do this :-)

File: src/segment_brewfather.go (hunk 2 above)
Comment (memcpy-rand-rand-rand) [reply src/segment_brewfather.go]: I pondered this one for a bit - figured a quick coded version that might be the golden path for many would be more efficient that trying to replicate with template logic.  Mainly coming at it from a perf angle, but I haven't profiled it.

It does feel a bit janky to have both.

I'll switch to default template for consistency.

File: src/segment_brewfather.go (hunk 1 above)
Comment (memcpy-rand-rand-rand) [reply src/segment_brewfather.go]: fixed and pushed.

File: src/segment_brewfather.go (hunk 2 above)
Comment (memcpy-rand-rand-rand) [reply src/segment_brewfather.go]: fixed and pushed

File: src/environment.go (hunk 3 above)
Comment (memcpy-rand-rand-rand) [reply src/environment.go]: done - pushed.

File: src/segment_brewfather.go (hunk 1 above)
Comment (memcpy-rand-rand-rand) [reply src/segment_brewfather.go]: on that topic - I notice some spacing misbehaviours depending on the glyph, are you tracking issues on that already?

Edit: never mind, it's our old friend the Terminal bug.

**Pull request merged**